src/tests/e2etest/E2ETest.cs

```diff
@@ -1,294 +1,375 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
-using System.Linq;
-using NUnit.Framework;
 using Microsoft.Azure.Devices;
 using Microsoft.Azure.Devices.Shared;
-using Microsoft.Extensions.Logging;
-using System.Threading.Tasks;
+using NUnit.Framework;
+using System;
 using System.Text.Json;
 using System.Text.RegularExpressions;
+using System.Threading.Tasks;
 
 namespace E2eTesting
 {
     [TestFixture]
-    public class E2eTest
+    public abstract class E2ETest
     {
-        // Local testing: populate and uncomment string below
-        // private string iotHubConnectionString = "HostName=<iot-hub>.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=<secret>";
-        private string iotHubConnectionString = Environment.GetEnvironmentVariable("E2E_OSCONFIG_IOTHUB_CONNSTR");
-        // Local testing: populate and uncomment the string below
-        // private string _deviceId = "ubuntu2004";
-        private string deviceId = Environment.GetEnvironmentVariable("E2E_OSCONFIG_DEVICE_ID");
-        private readonly string moduleId = "osconfig";
-        protected int twinTimeoutSeconds = 0;
-
-        // 30 minutes
-        private int twinTimeoutSecondsDefault = 60 * 30;
-
-        // 0.5 seconds
-        public readonly int twinRefreshIntervalMs = 500;
-
-        private ServiceClient _serviceClient;
-        protected RegistryManager _registryManager;
-        // vars needed to upload to blob store
-        private string sas_token = null;
-        private string upload_url = null;
-        private string resource_group_name = null;
-
-        private Twin twin;
-
-        public String DeviceId
-        {
-            get { return deviceId; }
-        }
+        private RegistryManager _registryManager;
 
-        public void SetTwinTimeoutSeconds(int seconds)
+        private readonly string _iotHubConnectionString = Environment.GetEnvironmentVariable("E2E_OSCONFIG_IOTHUB_CONNSTR")?.Trim('"');
+        private readonly string _moduleId = "osconfig";
+        private readonly string _deviceId = Environment.GetEnvironmentVariable("E2E_OSCONFIG_DEVICE_ID");
+
+        private readonly string _sasToken = Environment.GetEnvironmentVariable("E2E_OSCONFIG_SAS_TOKEN");
+        private readonly string _uploadUrl = Environment.GetEnvironmentVariable("E2E_OSCONFIG_UPLOAD_URL")?.Trim('\"');
+        private readonly string _resourceGroupName = Environment.GetEnvironmentVariable("E2E_OSCONFIG_RESOURCE_GROUP_NAME")?.Trim('\"');
+
+        private const int POLL_INTERVAL_MS = 1000;
+        private const int MAX_WAIT_TIME_SECONDS = 90;
+
+        protected const int ACK_SUCCESS = 200;
+        protected const int ACK_ERROR = 400;

```

Comment:
Can we expand these to treat any 2xx value as success and and any 4xx value as error?

Reply:
For this I recommend we keep specific values since each test expects a specific ack code depending on value that is set. Also it is the responsibility of each individual test to check this value, so if a range is expected, the test must validate the returned ack code against the range. In the tests we have now, all are expecting a specific value.